### PR TITLE
OMIM alignment - January 2024

### PR DESCRIPTION
**Pull request by sabrinatoro**
Addresses #7211 

(1st commit includes removal of some omim x-ref)

### CODE REVIEW

**Comment by sabrinatoro** [comment]
@nicolevasilevsky, this PR is ready for review. There were many QC to address, but I got almost all of them. 

I am still having a QC issue: FAIL Rule ../sparql/qc/mondo/qc-omimps-should-be-inherited.sparql: 16 violation(s)
I understand the error that is reported: terms with OMIMPS equivalent that do not have a "has_characteristic some inherited" axiom
BUT: the terms reported as errors DO HAVE these axioms from their parents. 
Should these axioms be asserted instead of inferred? Is that the problem? (tagging @matentzn too). 
Thanks!

**Comment by matentzn** [comment]
You can try to update the SPARQL query

```
FILTER NOT EXISTS {
    ?entity rdfs:subClassOf* [ rdf:type owl:Restriction ;
    owl:onProperty <http://purl.obolibrary.org/obo/RO_0000053> ;
    owl:someValuesFrom <http://purl.obolibrary.org/obo/MONDO_0021152> ] .
  }
```

By adding that `rdfs:subClassOf*` `*` expression and see if it works - not sure how SPARQL fares with subclass of anonymous expressions. 

Generally, there is a lot of redundancy in mondo-edit.obo, and there is no problem with that. During the release, the redundant axioms will be removed again. So either way is fine: changing the qc, and adding the assertion everywhere!

File: src/ontology/mondo-edit.obo
Comment (nicolevasilevsky) [inline src/ontology/mondo-edit.obo:135426]: if you turn this into the OMIMPS rather than the OMIM term, what happened to the OMIM term? did you check that all of the xrefs are equivalent to the OMIMPS and not OMIM:109400? This is normally a case where we would split this term (or [bud it](https://mondo.readthedocs.io/en/latest/editors-guide/splitting-classes/#splitting-a-more-generic-term-into-a-more-specific-terms_1))

File: src/ontology/mondo-edit.obo
Comment (nicolevasilevsky) [inline src/ontology/mondo-edit.obo:158083]: I think this term needs to be split b/c you are changing the meaning here

File: src/ontology/mondo-edit.obo
Comment (nicolevasilevsky) [inline src/ontology/mondo-edit.obo:202103]: I think this term needs to be split too.

Comment (nicolevasilevsky) [reply src/ontology/mondo-edit.obo:202103]: We don't want to lose the OMIMM classes, like OMIM:259680. And these xrefs are not all equiv to the OMIMPS

File: src/ontology/mondo-edit.obo
Comment (nicolevasilevsky) [inline src/ontology/mondo-edit.obo:248863]: I don't know that it is worthwhile to review every instance here- these probably need the same action as above

File: src/ontology/mondo-edit.obo
Comment (nicolevasilevsky) [inline src/ontology/mondo-edit.obo:262413]: this cannot be equiv to a proper OMIM term and an OMIMPS

File: src/ontology/mondo-edit.obo
Comment (sabrinatoro) [reply src/ontology/mondo-edit.obo:135426]: In all these cases, here OMIM:109400, a new term will be created during the next omim slurp/alignment, which I am planning on making as soon as this PR is merged, and before we release. So these OMIM terms will not be missing from Mondo from a user perspective. 

The reasoning behind it is that all of these new OMIMPS were created because there is a type 1 (previous OMIM) and type 2. So I want to make sure that we have both types instead of having a version where we have only 1 type (and having to add "single child" exclude from QC )

Comment (sabrinatoro) [reply src/ontology/mondo-edit.obo:135426]: Another way to see this is that this term would be split, but all the other x-refs and term ID would be maintained because it is referring to the higher concept. The "new term" in the split has not been created yet. 
(note: I am doing it this way only because I know that I will run the pipeline and get these "new terms" before the release. I wouldn't do it otherwise).

File: src/ontology/mondo-edit.obo
Comment (sabrinatoro) [reply src/ontology/mondo-edit.obo:202103]: I reviewed the x-refs, and all of them refer to the more generic term (ie not associated with a gene)

File: src/ontology/mondo-edit.obo
Comment (sabrinatoro) [reply src/ontology/mondo-edit.obo:262413]: This one was missed.

Comment (sabrinatoro) [reply src/ontology/mondo-edit.obo:262413]: updated according to my logic above.

File: src/ontology/mondo-edit.obo
Comment (sabrinatoro) [reply src/ontology/mondo-edit.obo:135426]: All of these terms are reported in [this spreadsheet-Sheet1](https://docs.google.com/spreadsheets/d/1A7ILok6Ae7mHQPD88E8k_dGxqLcKcqKWmN0hXpeItXU/edit#gid=0), in yellow, with the "check after next round" follow-up.

Comment (nicolevasilevsky) [reply src/ontology/mondo-edit.obo:135426]: If I am understanding correctly, you will be changing the ID of the term that is equivalent to OMIM:109400, and MONDO:0007187 will no longer mean the disease that has a variation in a specific gene but to a group of diseases. It looks like this term conflates the grouping class and subclass (there are two DO xrefs). I think the correct approach here is to split this class, I personally don't agree with your proposed approach.

Comment (nicolevasilevsky) [reply src/ontology/mondo-edit.obo:135426]: I'd recommend consulting with Chris on this.

Comment (sabrinatoro) [reply src/ontology/mondo-edit.obo:135426]: Currently, the concept for MONDO:0007187 (and other similar examples) already represents a group of diseases and does NOT represent a specific disease associated with a variation in a gene. 
In all these examples, if we were to do in manually, we would keep the ID for the general term and create a new term for the specific disease represented by the OMIM id. 
I am doing the same thing here, except that the creation of the new, more specific term will be added in the next PR. 

@cmungall can you check this discussion and share your thoughts? 
If we need to discuss it on a call, that's also ok, but please let us know. 

(note: depending on the timing of the discussion and the decision, the OMIM ingest might need to wait until the next release)

Comment (nicolevasilevsky) [reply src/ontology/mondo-edit.obo:135426]: > Currently, the concept for MONDO:0007187 (and other similar examples) already represents a group of diseases and does NOT represent a specific disease associated with a variation in a gene.

some of the equiv xrefs refer to the disease with the gene variation though:
- https://www.ebi.ac.uk/ols4/ontologies/doid/classes?obo_id=DOID%3A0070365
- https://www.ebi.ac.uk/ols4/ontologies/ncit/classes?short_form=NCIT_C2892

Comment (nicolevasilevsky) [reply src/ontology/mondo-edit.obo:135426]: > In all these examples, if we were to do in manually, we would keep the ID for the general term and create a new term for the specific disease represented by the OMIM id.

in the past, we have always obsoleted this term and created two new terms with two new IDs (split)

Comment (nicolevasilevsky) [reply src/ontology/mondo-edit.obo:135426]: the split process is definitely tedious. If we move away from it, I think we should definitely inform users about what is happening though because if someone is using the ID MONDO:0007187 with the understanding that it is equivalent to NCIT:C2892, which is the more specific term, then their annotation would become wrong.

Comment (sabrinatoro) [reply src/ontology/mondo-edit.obo:135426]: I think we need to discuss on a call, because I disagree with the statement "ID MONDO:0007187 with the understanding that it is equivalent to NCIT:C2892" (and I also disagree that NCIT:C2892 id actually more specific, but that's a different discussion). 

IMO, this is a case of ["Splitting a more generic term into a more specific term(s)"](https://mondo.readthedocs.io/en/latest/editors-guide/splitting-classes/#splitting-a-more-generic-term-into-a-more-specific-terms_1), and therefore we have already been doing this. 
This is based on the fact that the ontology term = the concept=the definition, and not the mappings. 
If we follow this logic, than we would have to obsolete Mondo terms every time we have incorrect mappings.
We do obsolete terms when the concept itself (definitions and axioms) was confusing. But I don't think the concept in these cases was confusing. 

I am not worried about the split process being tedious. But this means that we will not have an OMIM update for another 2 months.

Comment (nicolevasilevsky) [reply src/ontology/mondo-edit.obo:135426]: Sounds good. Good point about ontology term = the concept=the definition, this makes sense to me. There might be another case below where a logical axiom was deleted though, I'll review that and comment below.

File: src/ontology/mondo-edit.obo
Comment (nicolevasilevsky) [reply src/ontology/mondo-edit.obo:158083]: in this case, if we remove the axiom here, this changes the meaning of this term though, right?

**Review by nicolevasilevsky — APPROVED** [review]

**Review by twhetzel — APPROVED** [review]
approving based on discussion at curation call

**Pull request merged**
